### PR TITLE
8325680: Uninitialised memory in deleteGSSCB of GSSLibStub.c:179

### PR DESCRIPTION
**Pull request by wangweij**
Add a `memset` after memory is successfully allocated.

No regression test, hard to write without a KDC.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8325680](https://bugs.openjdk.org/browse/JDK-8325680): Uninitialised memory in deleteGSSCB of GSSLibStub.c:179 (**Bug** - P3)

### Reviewers
 * [Valerie Peng](https://openjdk.org/census#valeriep) (@valeriepeng - **Reviewer**) ⚠️ Review applies to [9983d06b](https://git.openjdk.org/jdk/pull/18015/files/9983d06b70471490bc59125930103e2a86c39524)
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - **Reviewer**) ⚠️ Review applies to [9983d06b](https://git.openjdk.org/jdk/pull/18015/files/9983d06b70471490bc59125930103e2a86c39524)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18015/head:pull/18015` \
`$ git checkout pull/18015`

Update a local copy of the PR: \
`$ git checkout pull/18015` \
`$ git pull https://git.openjdk.org/jdk.git pull/18015/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18015`

View PR using the GUI difftool: \
`$ git pr show -t 18015`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18015.diff">https://git.openjdk.org/jdk/pull/18015.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18015#issuecomment-1966677217)

### CODE REVIEW

**Comment by mcpowers** [comment]
Why do you have to set memory to zero?
If you really need to, `calloc()` sets memory to zero.

**Review by valeriepeng — APPROVED** [review]

**Comment by mcpowers** [comment]
Thanks for connecting the dots.

File: src/java.security.jgss/share/native/libj2gss/GSSLibStub.c [hunk 1]
```diff
@@ -196,7 +196,7 @@ gss_channel_bindings_t newGSSCB(JNIEnv *env, jobject jcb) {
     return GSS_C_NO_CHANNEL_BINDINGS;
   }
 
-  cb = malloc(sizeof(struct gss_channel_bindings_struct));
+  cb = calloc(1, sizeof(struct gss_channel_bindings_struct));
```
Comment (rhalade) [inline src/java.security.jgss/share/native/libj2gss/GSSLibStub.c:199]: With this update, memset calls at line 220 are unnecessary.

File: src/java.security.jgss/share/native/libj2gss/GSSLibStub.c [hunk 2]
```diff
@@ -210,15 +213,10 @@ gss_channel_bindings_t newGSSCB(JNIEnv *env, jobject jcb) {
 
   if ((*env)->IsInstanceOf(env, jcb, tlsCBCl)) {
       // TLS Channel Binding requires unspecified addrtype=0
-      cb->initiator_addrtype = GSS_C_AF_UNSPEC;
```
Comment (rhalade) [inline src/java.security.jgss/share/native/libj2gss/GSSLibStub.c:213]: Thanks for the update. I suggest not to remove this assignment as it helps understand the flow. Or if you prefer it then invert if condition to cover else part and avoid empty if block.

**Pull request closed without merging**